node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -167,6 +175,7 @@ impl ChainScraper {
 		// Clean up finalized:
 		for finalized_candidate in finalized.into_values().flatten() {
 			self.included_candidates.remove(&finalized_candidate);
+			self.backed_candidates.remove(&finalized_candidate);
```

Comment:
We might still be leaking memory (slowly). In case we saw an abandoned fork, where a candidate got backed but never included then we won't have an entry in `candidates_by_block_number`, hence we never remove the candidate from `backed_candidates`. 

I am also just realizing that even the already existing solution is not 100% correct: It could happen that a candidate gets included at different block heights on different forks, which could lead us to remove the candidate too early. We need to replace this pruning with a mechanism that makes sure that we only remove the entries once the highest seen block having relevant candidate events gets finalized.

Reply:
This is fixed with the latest changes.